spec/support/authentication_helpers.rb

```diff
@@ -9,4 +9,8 @@ def mock_successful_authentication(uid: "12345", name: "Alex", email: "alex@exam
       }
     )
   end
+
+  def stub_authenticated_session(name: "Alex", email: "alex@example.com")
```

Comment:
I'm unsure what the difference between `mock_successful_authentication` and `stub_authenticated_session` is here? Do we need both? Could they be combined?

Reply:
As I understand it `mock_successful_authentication` will allow us to test the omniauth callback phase. This essentially fakes the request phase and makes our fake response to our callback endpoint where we set a session and redirect.

I think this is useful to test in this detail on our authentication feature specs 👍 

So long as we have those feature specs providing this coverage, I think we can remove this overhead from every other test to make things run a bit quicker. These may mount up quickly when we have hundreds of test. In order to do test everything else I think we can stop testing the omniauth handshake each time by setting up a trusted session ourselves using `stuf_authenticated_session`. This is then checked by `logged_in_using_omniauth?` when we visit a particular page in a feature test and allowed through.